nilearn/plotting/displays.py

```diff
@@ -237,7 +245,24 @@ def transform_to_2d(self, data, affine):
                 The affine of the volume
 
         """
-        max_axis = 'xyz'.index(self.direction)
+        if self.direction in 'xlr':
+            max_axis = 0
+        else:
+            max_axis = '.yz'.index(self.direction)
+
+        # set unselected brain hemisphere activations to 0
+        if self.direction == 'l':
+            data_selection = data.copy()
+            x_center, _, _, _ = np.dot(np.linalg.inv(affine),
+                                       np.array([0, 0, 0, 1]))
+            data_selection[int(x_center):, :, :] = 0
```

Comment:
@AlexandreAbraham as I understood from http://nipy.org/nibabel/coordinate_systems.html, the affine is not necessarily diagonal but is invertible and its last row is 0,0,0,1. Nibabel documentation explain that [i,j,k,1] = dot(inv(M), [x,y,z,1]) so I am trusting thats the case. Since for the glass brain we are in MNI space I am assuming I can consider space coordinates 0,0,0 to ask for the actual voxel indices corresponding to the center.